pallets/connectors/src/message.rs

```diff
@@ -207,6 +207,24 @@ mod tests {
 			assert_eq!(msg.encode(), vec![0]);
 		}
 
+		#[test]
+		fn encoding_domain_evm() {
+			let evm_domain = Domain::EVM(u64::MAX);
```

Comment:
Maybe you can add a few more sample tests for common chain IDs, e.g. Ethereum Mainnet (1), Avalanche C-cHain (43114), Moonbeam (1284). That way the connectors solidity tests can easily implement the same tests and ensure it all matches.

Reply:
@offerijns how do we want to handle the different environments within Polkadot? Meaning, `Domain::Parachain(ParachianId::Moonbeam)`, following to your examples, should have different values if we are running this pallet on Moobase Alpha, Rococo, Kusama, or Polkadot because "Moonbeam" has a different parachain id on each of those chains.

My idea was to just index the parachains by a fixed enumerated index (0, 1, 2, etc) so that it's not chain dependant. That's why, as the test below now shows, `Domain::Parachain(ParachainId::Moonbeam)` encodes to `0x0100`.

What do you think?

Reply:
Actually just realized that we need to make the domain encode to a fixed-size output. Now the `Evm` and the `Parachain` variants encode to different sizes.

Reply:
> Actually just realized that we need to make the domain encode to a fixed-size output. Now the Evm and the Parachain variants encode to different lengths.

I don't think it really needs to be fixed length. In the Solidity code, we can check if the first byte represents an EVM or Polkadot domain. Then we either parse 8 more bytes in the EVM case or skip to the next parameter in the Polkadot case. This will lead to more efficient messages than zero padding it.

> My idea was to just index the parachains by a fixed enumerated index (0, 1, 2, etc) so that it's not chain dependant. That's why, as the test below now shows, Domain::Parachain(ParachainId::Moonbeam) encodes to 0x0100.

I think that makes sense!

Reply:
Perfect, thanks @offerijns :)